vector/src/main/java/im/vector/app/features/home/room/detail/views/RoomDetailLazyLoadedViews.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.detail.views
+
+import android.view.View
+import android.view.ViewStub
+import im.vector.app.core.ui.views.FailedMessagesWarningView
+import im.vector.app.databinding.FragmentRoomDetailBinding
+import im.vector.app.features.invite.VectorInviteView
+import kotlin.reflect.KMutableProperty0
+
+/**
+ * This is an holder for lazy loading some views of the RoomDetail screen.
+ * It's using some ViewStub where it makes sense.
+ */
+class RoomDetailLazyLoadedViews {
+
+    private var roomDetailBinding: FragmentRoomDetailBinding? = null
+
+    private var failedMessagesWarningView: FailedMessagesWarningView? = null
+    private var inviteView: VectorInviteView? = null
+
+    fun bind(roomDetailBinding: FragmentRoomDetailBinding) {
+        this.roomDetailBinding = roomDetailBinding
+    }
+
+    fun unBind() {
+        roomDetailBinding = null
+        inviteView = null
+        failedMessagesWarningView = null
+    }
+
+    fun failedMessagesWarningView(inflateIfNeeded: Boolean, callback: FailedMessagesWarningView.Callback? = null): FailedMessagesWarningView? {
+        return getOrInflate(inflateIfNeeded, roomDetailBinding?.failedMessagesWarningStub, this::failedMessagesWarningView)?.apply {
+            this.callback = callback
+        }
+    }
+
+    fun inviteView(inflateIfNeeded: Boolean): VectorInviteView? {
+        return getOrInflate(inflateIfNeeded, roomDetailBinding?.inviteViewStub, this::inviteView)
+    }
+
+    private inline fun <reified T : View> getOrInflate(inflateIfNeeded: Boolean, stub: ViewStub?, reference: KMutableProperty0<T?>): T? {
```

Comment:
This could be extracted to a more generic place

Reply:
Will see if we reuse this mechanism somewhere else I guess, no?

Reply:
Yes, but if it is hidden here, there is less chance. Anyway, ok